AppBuilder/platform/ABDataCollection.js

```diff
@@ -311,6 +311,15 @@ module.exports = class ABDataCollection extends ABDataCollectionCore {
          await Promise.all(pendingRelatedRuleDC);
       this._cond = cond;
       this._lock = new AB.app.utils.Lock();
+
+      // at the end of init(), I'm forcing the status = 0
+      // this should cause loadData() to go out and force
+      // load the data.
+      // @Guy: Is this needed? <----
+      //
+      await this._lock.acquire();
+      await this.AB.app.resources.storage.set(this.refStorage(), "status", "0");
```

Comment:
Everything looks good to me for the paging. For this one, I think it is not needed. I actually expect `loadData` to fetch all data from the server only once after we register the `authToken`. After that, it should load data from the storage even if we reload the page. New updated data will be tracked and synced from the server by `updateSyncData`, filtered by the `updated_at` column. However, it seems like `updated_at` is not being updated after I update any records on the server, based on my tests. So, for now, I forced `updateSyncData` to reload all data instead. The code is still in my local. I will push it soon.